src/controllers/posts.js

```diff
@@ -99,3 +100,25 @@ export const deletePost = async (req, res) => {
     return res.status(500).json({ error: 'Internal server error' });
   }
 };
+
+// update posts
+export const updatePost = async (req, res) => {
```

Comment:
no need to get the post id from the req params, already checked from postExists middleware. Use lodash get. also no need to check the post id is valid, already checked in middleware.